docs/resources/oxide_switch_port_settings.md

```diff
@@ -0,0 +1,282 @@
+---
+page_title: "oxide_switch_port_settings Resource - terraform-provider-oxide"
+---
+
+# oxide_switch_port_settings (Resource)
+
+This resource manages switch port settings for configuring network switch ports
+including addresses, BGP peers, interfaces, links, and routes.
+
+!> Switch port settings defined by this resource are considered exhaustive and
+will overwrite any other switch port settings upon apply.
+
+## Example Usage
+
+### Basic Switch Port Settings
+
+```hcl
+resource "oxide_switch_port_settings" "example" {
+  name        = "example"
+  description = "Switch port settings."
+
+  port_config = {
+    geometry = "qsfp28x1"
+  }
+
+  addresses = [
+    {
+      link_name = "phy0"
+      addresses = [
+        {
+          address        = "192.168.1.123/24"
+          address_lot_id = "38223e3a-76da-400d-a1e2-8cb4d242095a"
+        },
+      ]
+    },
+  ]
+
+  links = [
+    {
+      link_name = "phy0"
+      autoneg   = false
+      mtu       = 1500
+      speed     = "speed1_g"
+      lldp = {
+        enabled = false
+      }
+    },
+  ]
+
+  routes = [
+    {
+      link_name = "phy0"
+      routes = [
+        {
+          dst = "0.0.0.0/0"
+          gw  = "192.168.1.1"
+        },
+      ]
+    },
+  ]
+}
+```
+
```

Comment:
Thinking about it, It would definitely be nice to have an additional example with a more complex configuration as well.